test/System/Config.cpp

```diff
@@ -4,14 +4,6 @@
 
 TEST_CASE("SFML/Config.hpp")
 {
-    SUBCASE("Version macros")
-    {
-        CHECK(SFML_VERSION_MAJOR == EXPECTED_SFML_VERSION_MAJOR);
-        CHECK(SFML_VERSION_MINOR == EXPECTED_SFML_VERSION_MINOR);
-        CHECK(SFML_VERSION_PATCH == EXPECTED_SFML_VERSION_PATCH);
-        CHECK(SFML_VERSION_IS_RELEASE == EXPECTED_SFML_VERSION_IS_RELEASE);
```

Comment:
I'm guessing that you don't like how the implementation and these tests so closely resemble each other now. If so then let's find a better way to test the contents of Config.hpp instead of simply giving up on the idea of unit testing the API.

Reply:
Well, the rationale here is that the input is already given in the `project()` command and the `SFML_VERSION_IS_RELEASE` definition after that. I'm not sure what's there to test if the values here are now derived from the CMakeLists.txt file instead of being duplicated across multiple files.

Reply:
> I'm not sure what's there to test

The public API is what there is to test. We wouldn't put the version number in Config.hpp if we didn't plan on users using it. Because users are using it, we should seriously consider writing tests for it.

Reply:
I mean that the value of these macros is derived from the values in the CML file. I'm not even sure what a test for that would look like. The reason there was a test before is because the version was maintained separately in the header and the CML, so divergence was a possibility. Now, whatever goes in the CML also goes in the header.

Reply:
> The reason there was a test before is because the version was maintained separately in the header and the CML

No, the rationale was because I want to test as much of the public API as possible. 

> Now, whatever goes in the CML also goes in the header.

Unless someone modifies the .in file and breaks it in some way by either changing the name or value of those definitions. We can debate how likely that may be but the fact stands that we ought to test the library.

Reply:
OK, I understand now. So the test should be that the definitions exist and have a numeric value.

Reply:
Yes but preferably that they exist and have the _correct_ numeric value. Any tests are better than no tests though so I'm willing to meet you in the middle if you think the tests as they currently exist in `master` are overkill.

Reply:
I just find it overly defensive, but whatever. I just fixed the genex up and let it be.